pkg/client/vault/client.go

```diff
@@ -205,3 +304,95 @@ func compressZCAP(zcap *zcapld.Capability) (string, error) {
 
 	return base64.URLEncoding.EncodeToString(compressed.Bytes()), nil
 }
+
+func lastElm(s, sep string) string {
+	all := strings.Split(s, sep)
+
+	return all[len(all)-1]
+}
+
+func buildEDVURI(h, vid, did string) string {
+	return fmt.Sprintf("%s/encrypted-data-vaults/%s/documents/%s", h, vid, did)
+}
+
+func prepareRemoteJWEKey(keyURL string, kmsImpl kms.KeyManager) ([]byte, string, error) {
+	kid := lastElm(keyURL, "/")
+
+	pubKeyBytes, err := kmsImpl.ExportPubKeyBytes(kid)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to retrieve main public key bytes from remote KMS: %w", err)
+	}
+
+	return pubKeyBytes, kid, nil
+}
+
+func encryptKey(wKMS KeyManager, wCrypto crypto.Crypto, cek []byte) (string, error) {
+	_, kidURL, err := wKMS.Create(kms.NISTP256ECDHKW)
+	if err != nil {
+		return "", fmt.Errorf("create: %w", err)
+	}
+
+	pubKeyBytes, jweCryptoKID, err := prepareRemoteJWEKey(kidURL.(string), wKMS)
+	if err != nil {
+		return "", fmt.Errorf("prepare remote JWE key: %w", err)
+	}
+
+	ecPubKey := &crypto.PublicKey{KID: jweCryptoKID}
+
+	err = json.Unmarshal(pubKeyBytes, ecPubKey)
+	if err != nil {
+		return "", fmt.Errorf("unmarshal: %w", err)
+	}
+
+	encrypter, err := jose.NewJWEEncrypt(jose.A256GCM, jose.DIDCommEncType, "", nil,
+		[]*crypto.PublicKey{ecPubKey}, wCrypto)
+	if err != nil {
+		return "", fmt.Errorf("new JWE encrypt: %w", err)
+	}
+
+	jwe, err := encrypter.Encrypt(cek)
+	if err != nil {
+		return "", fmt.Errorf("encrypt: %w", err)
+	}
+
+	return jwe.FullSerialize(json.Marshal)
+}
+
+func encryptContent(content interface{}) ([]byte, []byte, error) {
+	cek := random.GetRandomBytes(crypto.DefKeySize)
```

Comment:
`crypto.DefKeySize` = 32, which means we're generating a random 256-bit key for encryption, which is fine, but I feel a little uneasy about importing the frameworks "default key size" constant. I suggest we define our own constant here... eg. `EncryptionKeySizeBytes`.

Reply:
So I believe you need two things:

* the webkms implementation of `KMS` (to create the keys and get the keyIDs back)
* the `JWEEncrypt` initialized with the webkms implementation of `Crypto` and the keyID/keyHandle

You'd just call `JWEEncrypt.Encrypt(plaintext []byte)` and it'll do the key wrapping for you.